packages/Python/lldbsuite/test/lang/swift/swift_interface_static/TestSwiftInterfaceStaticNoDebugInfo.py

```diff
@@ -0,0 +1,136 @@
+        self.build()
+        self.do_test()
+
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+
+
+    def do_test(self):
+        # The custom module cache location
+        mod_cache = self.getBuildArtifact("MCP")
+
+        # Clear the module cache (populated by the Makefile build)
+        shutil.rmtree(mod_cache)
+        self.assertFalse(os.path.isdir(mod_cache),
+                         "module cache should not exist")
+
+        # Update the settings to use the custom module cache location
+        self.runCmd('settings set symbols.clang-modules-cache-path "%s"' % mod_cache)
+
+        # Set a breakpoint in and launch the main executable
+        (target, process, thread, bkpt) = lldbutil.run_to_source_breakpoint(self,
+                "break here", self.main_source_spec,
+                exe_name="main")
+
+        self.frame = thread.frames[0]
+        self.assertTrue(self.frame, "Frame 0 is valid.")
+
+        # Check we are able to access the public fields of variables whose
+        # types are from the .swiftinterface-only modules
+        var = self.frame.FindVariable("x")
+        lldbutil.check_variable(self, var, False, typename="AA.MyPoint")
+
+        child_y = var.GetChildMemberWithName("y") # MyPoint.y is public
+        lldbutil.check_variable(self, child_y, False, value="0")
+
+        child_x = var.GetChildMemberWithName("x") # MyPoint.x isn't public
+        self.assertFalse(child_x.IsValid())
+
+        # Expression evaluation using types from the .swiftinterface only
+        # modules should work too
+        lldbutil.check_expression(self, self.frame, "y.magnitudeSquared", "404", use_summary=False)
+        lldbutil.check_expression(self, self.frame, "MyPoint(x: 1, y: 2).magnitudeSquared", "5", use_summary=False)
+
+        # Check the module cache was populated with the .swiftmodule files of
+        # the loaded modules
+        self.assertTrue(os.path.isdir(mod_cache), "module cache exists")
+        a_modules = glob.glob(os.path.join(mod_cache, 'AA-*.swiftmodule'))
+        b_modules = glob.glob(os.path.join(mod_cache, 'BB-*.swiftmodule'))
+        c_modules = glob.glob(os.path.join(mod_cache, 'CC-*.swiftmodule'))
+        self.assertEqual(len(a_modules), 1)
+        self.assertEqual(len(b_modules), 1)
+        self.assertEqual(len(c_modules), 0)
+
+        # Update the timestamps of the modules to a time well in the past
+        for file in a_modules + b_modules:
+            make_old(file)
+
+        # Re-import module A and B
+        self.runCmd("expr import AA")
+        self.runCmd("expr import BB")
+
+        # Import C for the first time
+        self.runCmd("expr import CC")
+
+        # Not clear why this doesn't work here, when it does with a dylib...
+        lldbutil.check_expression(self, self.frame, "Baz.baz()", None, use_summary=False)
```

Comment:
Could you please file a bug report for this?

Reply:
Static libraries just can't be loaded dynamically; it's not related to this new feature.